tests/cases/fourslash/completionAfterQuestionDot.ts

```diff
@@ -1,20 +1,43 @@
 /// <reference path="fourslash.ts" />
 // @strict: true
 
-//// interface User {
+//// class User {
+////     #foo: User;
+////     bar: User;
 ////     address?: {
 ////         city: string;
 ////         "postal code": string;
+////     };
+////     constructor() {
+////         this.address[|?./*1*/|];
+////         this[|?./*2*/|];
+////         this?.bar[|?./*3*/|];
 ////     }
 //// };
-//// declare const user: User;
-//// user.address[|?./**/|]
 
 verify.completions({
-    marker: "",
+    marker: "1",
     exact: [
         { name: "city", text: "(property) city: string" },
         { name: "postal code", text: "(property) \"postal code\": string", insertText: "?.[\"postal code\"]", replacementSpan: test.ranges()[0] }
     ],
     preferences: { includeInsertTextCompletions: true },
 });
+
+verify.completions({
+    marker: "2",
+    exact: [
+        { name: "bar" },
+        { name: "address" }
+    ],

```

Comment:
Nice work on this, it's cool that you updated the completions so they don't suggest private identifiers where they wouldn't be valid.